ggml.c

```diff
@@ -1523,6 +1523,214 @@ static inline void __sse_f16x4_store(ggml_fp16_t *x, __m128 y) {
 #define GGML_F16_VEC_MUL             GGML_F32Cx4_MUL
 #define GGML_F16_VEC_REDUCE          GGML_F32Cx4_REDUCE
 
+#elif defined(__loongarch_asx)
+
+#define GGML_SIMD
+
+// F32 LASX
+
+typedef union
+{
+    int32_t i;
+    float f;
+} FloatInt;
+/* float type data load instructions */
+static __m128 __lsx_vreplfr2vr_s(float val)
+{
+    FloatInt fi_tmpval = {.f = val};
+    return (__m128)__lsx_vreplgr2vr_w(fi_tmpval.i);
+}
+
+static __m256 __lasx_xvreplfr2vr_s(float val)
+{
+    FloatInt fi_tmpval = {.f = val};
+    return (__m256)__lasx_xvreplgr2vr_w(fi_tmpval.i);
+}
+#define GGML_F32_STEP 32
+#define GGML_F32_EPR  8
```

Comment:
Style:

- opening braces on same line
- snake_case

```suggestion
typedef union {
    int32_t i;
    float f;
} ggml_fi32;

/* float type data load instructions */
static __m128 __lsx_vreplfr2vr_s(float val) {
    ggml_fi32 fi_tmpval = {.f = val};
    return (__m128)__lsx_vreplgr2vr_w(fi_tmpval.i);
}

static __m256 __lasx_xvreplfr2vr_s(float val) {
    ggml_fi32 fi_tmpval = {.f = val};
    return (__m256)__lasx_xvreplgr2vr_w(fi_tmpval.i);
}

#define GGML_F32_STEP 32
#define GGML_F32_EPR  8
```

Reply:
We tried to inference about some models, e.g. ggml-model-q4_0. can get similar results to X86.

I'll modify the code style again, and by the way, I'll ask if there's a file like .clang-format or format script so I can easily modify the code style

Reply:
Sorry, we don't have a `.clang-format` - the only way is to look around nearby code and match the existing patterns